flink-optimizer/src/main/java/org/apache/flink/optimizer/traversals/RangePartitionRewriter.java

```diff
@@ -113,10 +113,12 @@ public void postVisit(PlanNode node) {
 					throw new InvalidProgramException("Range Partitioning not supported within iterations.");
```

Comment:
range partitioning is possible if the data distribution is provided. Please refactor this code accordingly.